arangod/AsyncRegistryServer/RestHandler.cpp

```diff
@@ -46,8 +48,18 @@ auto all_undeleted_promises()
   std::vector<Id> roots;
   registry.for_promise([&](PromiseSnapshot promise) {
     if (promise.state != State::Deleted) {
-      forest.insert(promise.id, promise.waiter, promise);
-      if (promise.waiter == nullptr) {
+      // only async waiters are considered here, they point to other promises in
+      // the registry
+      // sync waiters point to threads that are waiting synchronously and are
+      // not relevant for creating the forest
+      auto waiter =
+          promise.waiter.has_value()
+              ? (std::holds_alternative<AsyncWaiter>(promise.waiter.value())
+                     ? std::get<AsyncWaiter>(promise.waiter.value())
+                     : nullptr)
+              : nullptr;
```

Comment:
[There](https://en.cppreference.com/w/cpp/utility/variant/get_if) is `std::get_if`. It is a bit wired because you have to put a pointer to the variant, but you get a pointer to `T` which is `nullptr` is the `T` is not in the variant. This allows you to write code like
```
if (auto t = std::get_if<T>(&variant); t) {
   // now just use t as pointer.
}
```

Reply:
True, but with all options in the variant I can now just do a visit, which is even better readable.